### PR TITLE
Add PyVista examples!

### PR DESCRIPTION
**Pull request by banesullivan**
Finally getting around to https://github.com/sphinx-gallery/sphinx-gallery/issues/441#issuecomment-648133448

I thought it'd be cool to show of some of PyVista's examples here and point folks over to what we've built using Sphinx-Gallery -- it's such an awesome Sphinx extension!!

Hopefully adding this isn't a burden on the CI since you're already supporting Mayavi, but we'll see....

cc @akaszynski

Preview:

Screen Shot 2021-12-02 at 12 20 09 AM

Screen Shot 2021-12-02 at 12 20 21 AM

### CODE REVIEW

**Comment by banesullivan** [comment]
Hmm... not sure what this error is all about, but the PyVista examples are completing!

```
Exception occurred:
  File "/home/circleci/.local/lib/python3.8/site-packages/sphinx/setup_command.py", line 177, in run
    raise DistutilsExecError(
distutils.errors.DistutilsExecError: caused by html builder.
The full traceback has been saved in /tmp/sphinx-err-6x14p13k.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!

Exited with code exit status 1
CircleCI received exit code 1
```

File: pyvista_examples/plot_glyphs.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+"""
+
+import numpy as np
+
+# sphinx_gallery_thumbnail_number = 4
+import pyvista as pv
+from pyvista import examples
+
+###############################################################################
+# Glyphying can be done via the ``pyvista.DataSetFilters.glyph`` filter
```
Comment (lucyleeow) [inline pyvista_examples/plot_glyphs.py:17]: would this work?
```suggestion
# Glyphying can be done via the :func:`pyvista.DataSetFilters.glyph` filter
```

File: pyvista_examples/plot_collisions.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+import numpy as np
+
+import pyvista as pv
+
+pv.set_plot_theme("document")
+
+
+###############################################################################
+# Create the main mesh and the secondary "moving" mesh
```
Comment (lucyleeow) [inline pyvista_examples/plot_collisions.py:38]: nit
```suggestion
# Create the main mesh and the secondary "moving" mesh.
```

File: pyvista_examples/plot_collisions.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
+"""
+.. _collision_example:
+
+Collision
+~~~~~~~~~
+Perform a collision detection between two meshes.
+
+This example use the ``collision`` filter to detect the faces from one sphere
```
Comment (lucyleeow) [inline pyvista_examples/plot_collisions.py:8]: nit
```suggestion
This example uses the ``collision`` filter to detect the faces from one sphere
```

**Review by lucyleeow — APPROVED** [review]
Thanks for adding this great example. LGTM (some small nits but happy to merge as is) - do you want to take a look too @larsoner ?

File: pyvista_examples/README.rst [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+PyVista Examples
+----------------
+
+Examples from the Sphinx-Gallery using PyVista for embedding 3D plots.
+`Learn more about PyVista <https://docs.pyvista.org>`_ and see their
+`extensive examples gallery <https://docs.pyvista.org/examples/index.html>`_
+built with Sphinx-Gallery!
+
+Take a look at the :ref:`image_scrapers` section to learn more about using
+PyVista in your example gallery.
```
Comment (larsoner) [inline pyvista_examples/README.rst:18]: I think it's worth mentioning explicitly that it is *not* enabled by default.

File: pyvista_examples/plot_glyphs.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+"""
+
+import numpy as np
+
+# sphinx_gallery_thumbnail_number = 4
+import pyvista as pv
+from pyvista import examples
+
+###############################################################################
+# Glyphying can be done via the :func:`pyvista.DataSetFilters.glyph` filter
```
Comment (larsoner) [inline pyvista_examples/plot_glyphs.py:17]:
```
/home/circleci/project/doc/auto_pyvista_examples/plot_glyphs.rst:48: WARNING: py:func reference target not found: pyvista.DataSetFilters.glyph
```
So I did:
```
python -m sphinx.ext.intersphinx https://docs.pyvista.org/objects.inv > pyvista.txt
```
Looks like it's a `meth`

```suggestion
# Glyphying can be done via the :meth:`pyvista.DataSetFilters.glyph` filter
```

**Review by larsoner — COMMENTED** [review]
Otherwise LGTM!

**Comment by larsoner** [comment]
Nice!

https://2442-25860190-gh.circle-artifacts.com/0/rtd_html/auto_pyvista_examples/index.html

File: pyvista_examples/README.rst (hunk 4 above)
Comment (banesullivan) [reply pyvista_examples/README.rst]: Beat me to it, thanks!

**Comment by banesullivan** [comment]
I just noticed the "launch on MyBinder" button - could we make sure `ipyvtklink` is installed in that environment for in-notebook plotting?

**Comment by larsoner** [comment]
Probably -- feel free to do it in this PR, otherwise I'm happy to merge this as-is and deal with `binder` in another PR

**Comment by banesullivan** [comment]
Eh, we can do this later - I don't want to add PyVista+ipyvtklink directly to the requirements.txt just yet

**Pull request merged**